pkg/controller/cloudprivateipconfig/cloudprivateipconfig_controller.go

```diff
@@ -179,24 +179,98 @@ func (c *CloudPrivateIPConfigController) SyncHandler(key string) error {
 		return err
 	}
 
-	// At most one of nodeToAdd or nodeToDel will be set
-	nodeToAdd, nodeToDel := c.computeOp(cloudPrivateIPConfig)
+	// At most one of nodeNameToAdd or nodeNameToDel will be set
+	nodeNameToAdd, nodeNameToDel := c.computeOp(cloudPrivateIPConfig)
 	switch {
 	// Dequeue on NOOP, there's nothing to do
-	case nodeToAdd == "" && nodeToDel == "":
+	case nodeNameToAdd == "" && nodeNameToDel == "":
 		return nil
-	case nodeToDel != "":
-		klog.Infof("CloudPrivateIPConfig: %q will be deleted from node: %q", key, nodeToDel)
+	case nodeNameToAdd != "" && nodeNameToDel != "":
+		klog.Infof("CloudPrivateIPConfig: %q will be moved from node %q to node %q", key, nodeNameToDel, nodeNameToAdd)
+		nodeToDel, err := c.nodesLister.Get(nodeNameToDel)
+		if err != nil {
+			return err
+		}
+		nodeToAdd, err := c.nodesLister.Get(nodeNameToAdd)
+		if err != nil {
+			return err
+		}
+
+		status = &cloudnetworkv1.CloudPrivateIPConfigStatus{
+			Node: nodeNameToDel,
+			Conditions: []metav1.Condition{
+				{
+					Type:               string(cloudnetworkv1.Assigned),
+					Status:             metav1.ConditionUnknown,
+					ObservedGeneration: cloudPrivateIPConfig.Generation,
+					LastTransitionTime: metav1.Now(),
+					Reason:             cloudResponseReasonPending,
+					Message:            "Moving IP address",
+				},
+			},
+		}
+		if cloudPrivateIPConfig, err = c.updateCloudPrivateIPConfigStatus(cloudPrivateIPConfig, status); err != nil {
+			return fmt.Errorf("error updating CloudPrivateIPConfig: %q during move operation, err: %v", key, err)
+		}
 
-		node, err := c.nodesLister.Get(nodeToDel)
+		// This is a blocking call. If the IP is not assigned then don't treat
+		// it as an error.
+		var i interface{} = c.cloudProviderClient
+		withMover, ok := i.(cloudprovider.CloudProviderWithMoveIntf)
```

Comment:
nit: I find the below way easier to read, but that may be personal preference:
~~~
withMover, ok := c.cloudProviderClient.(cloudprovider.CloudProviderWithMoveIntf)
~~~